fnConsensus/reactor.go

```diff
@@ -412,6 +401,8 @@ func (f *FnConsensusReactor) vote(fnID string, fn Fn, currentValidators *types.V
 		return
 	}
 
+	f.state.Message[fnID] = message
+
```

Comment:
This needs to be moved down, after `f.stateMtx.Lock()` since you need to hold the lock before updating the state.

Reply:
yeah, wrong line